server/src/main/scala/actors/RecoveryActorLike.scala

```diff
@@ -58,6 +59,27 @@ trait RecoveryActorLike extends PersistentActor with RecoveryLogging {
     }
   }
 
+  def persistAndMaybeSnapshotWithAck(messageToPersist: GeneratedMessage, replyTo: ActorRef, ackMsg: Any): Unit = {
+    persist(messageToPersist) { message =>
+      val messageBytes = message.serializedSize
+      log.info(s"Persisting $messageBytes bytes of ${message.getClass}")
+
+      message match {
+        case m: AnyRef =>
+          context.system.eventStream.publish(m)
+          bytesSinceSnapshotCounter += messageBytes
+          messagesPersistedSinceSnapshotCounter += 1
+          logCounters(bytesSinceSnapshotCounter, messagesPersistedSinceSnapshotCounter, snapshotBytesThreshold, maybeSnapshotInterval)
+
+          snapshotIfNeeded(stateToMessage)
+
+          replyTo ! ackMsg
+        case _ =>
+          log.error("Message was not of type AnyRef and so could not be persisted")
```

Comment:
should we be logging what the message is if we get one that can't be perissted? Seems like a bug if that happens.

Reply:
Yeah this is actually redundant old code that will never execute. I've tidied it up now and consolidated the two persistAndMaybeSnapshot functions into one